src/item.cpp

```diff
@@ -4621,16 +4623,55 @@ std::string item::tname( unsigned int quantity, bool with_prefix, unsigned int t
                   contents_item.charges > 1 )
                 ? contents_item.charges
                 : 1;
+
+            // Color second half of container contents depending on perishability (highest perishability of all contents)
+            if( contents_item.is_food() ) {
+                // For comestibles
+				const item_pocket* const parent_pocket = contained_where(contents_item);
+                colorprefix = "<" + string_from_color( contents_item.color_in_inventory() ) + ">";
+                colorprefix.replace( 1, 1, "color" ); // changes <c_cyan> to <color_cyan>
+                if( contents.get_sealed_summary() == item_contents::sealed_summary::all_sealed && parent_pocket->spoil_multiplier() == 0 ) {
+                    colorprefix = "<color_light_blue>";
+                }
+                colorsuffix = "</color>";
+            }
+
             contents_suffix_text = string_format( pgettext( "item name",
                                                   //~ [container item name] " > [inner item  name]"
-                                                  " > %1$s" ),
+                                                  " > %1$s%2$s" ),
+                                                  /* with_contents=false for nested items to prevent excessively long names */
+                                                  colorprefix, contents_item.tname( contents_count, true, 0, false )
                                                   /* with_contents=false for nested items to prevent excessively long names */
-                                                  contents_item.tname( contents_count, true, 0, false ) );
+                                                ) + string_format( "%s", colorsuffix );
```

Comment:
Please store the color instead of the color prefix/suffix strings and use `colorize( text, color )` (from `color.h`) to add color to the text.

Reply:
Oh! I was not aware of that function, but will use it now!